push-to-s3-and-invalidate-cloudfront/action.yml

```
@@ -0,0 +1,148 @@
+
+inputs:
+  build-directory:
+    description: The directory that contains the build files to upload to S3
+    required: true
+  pull-request-number:
+    description: The pull request number for which a preview needs to be generated/updated
+    required: true
+  aws-cloudfront-distribution-id:
+    description: The cloudfront distribution ID for the app
+    required: true
+  aws-region:
+    description: The aws region used by the IAM machine user
+    required: true
+  aws-access-key-id:
+    description: The aws access key id for the IAM machine user
+    required:  true
+  aws-secret-access-key:
+    description: The aws secret access key for the IAM machine user
+    required:  true
+  s3-bucket:
+    description: The name of the S3 bucket
+    required: true
+
+  s3-action:
+    description: |
+      The action to take on the S3 bucket. Typically this should be `update` to push the latest files to S3 or `delete` to delete the folder from S3.
+      If this is not specified, the action will be infered from the $github.event.action variable
+    required: false
+  s3-dir-to-upload-to:
+    description: |
+      The S3 directory to upload the files to. If unspecified, the default `PR-$pull-request-number` will be used
+      or the directory may be infered from the github context. A value of `/` indicates the root of the bucket.
+    required: false
+    default: "pr-previews"
+  s3-dir-to-delete-from:
+    description: |
+      The directory on S3 whose contents need to be deleted. If unspecified, the default `PR-$pull-request-number` will be used
+      or the directory may be infered from the github context
+    required: false
+    default: "pr-previews"
+  s3-files-to-exclude:
+    description: |
+      A pattern or a comma separated list of files to exclude when running the aws s3 sync command. 
+      Refer to https://docs.aws.amazon.com/cli/latest/reference/s3/index.html#use-of-exclude-and-include-filters or more information
+      Set this variable to an empty string if you don't want to exlcude any files
+      If unspecified will default to `pr-previews/*`
+    required: false
+    default: "pr-previews/*"
+  invalidate-cloudfront-cache:
+    description: A flag to indicate if the cloudfront cache should be invalidated
+    required: false
+    default: "false"
+
+outputs:
+  dir_to_upload_to:
+    description: The S3 directory to upload the files to. Computed based on user input
+    value: ${{ steps.upload.dir_to_upload_to }}
+  dir_to_delete_from:
+    description: The S3 directory to upload the files to. Computed based on user input
+    value: ${{ steps.delete.dir_to_delete_from }}
+  files_to_exlude:
+    description: Files to exclude from S3 when using the sync command.
+    value: ${{ steps.exclude.files_to_exclude }}
+
+runs:
+  using: composite
+  steps:
+    - name: Compute the S3 directory to upload to
+      shell: bash
+      id: upload
+      run: |
+        if ${{ inputs.s3-dir-to-upload-to }} == "pr-previews"

```

Instead of doing it this way, would be possible to add a boolean like `is_pull_request_preview` (threw that name off of the top of head but a better name is welcome).

And then this line could be:
```suggestion
        if ${{ inputs.is_pull_request_preview }} == 'true'
```
This way if the folder you are attempting to put your pr-previews in is not called 'pr-previews', this would still work

---

Yeah bash syntax was lost while going back and forth with actions and bash. Fixed the conditional logic so no need for the `is_pull_request_preview` flag